src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/onetimecode/OneTimeCodeEmailSender.kt

```diff
@@ -62,18 +128,16 @@ class OneTimeCodeEmailSender(
 
     javaMailSender.createMimeMessage()
       .also { mimeMessage ->
-        MimeMessageHelper(mimeMessage, true).apply {
+        MimeMessageHelper(mimeMessage, MULTIPART_MODE_MIXED_RELATED).apply {
           setTo(email)
           setSubject("Your code for Send legal mail to prisons")
-          setText(textBody(oneTmeCode, codeExpiry), htmlBody(oneTmeCode, codeExpiry))
+          setText(textBodyTemplate(oneTmeCode, codeExpiry), htmlBodyTemplate(oneTmeCode, codeExpiry))
+          addInline("crest.png", ClassPathResource("email/crest.png"))
+          addInline("footer-image.png", ClassPathResource("email/footer-image.png"))
         }
       }
       .also { mimeMessage ->
         javaMailSender.send(mimeMessage)
       }
   }
-
-  private fun textBody(code: String, codeExpiry: String) = String.format(TEXT_BODY_TEMPLATE, code, codeExpiry)
-
-  private fun htmlBody(code: String, codeExpiry: String) = String.format(HTML_BODY_TEMPLATE, code, codeExpiry)
```

Comment:
I had to ditch String.format because it didn't like the semi-colons in the HTML. But good old Kotlin did the job.

Reply:
Nice. And to be fair I wish I'd have used kotlin's string interpolation in the first place! Its much nicer 👍